tests/passes/infer-share-pass/basic1.futil

```diff
@@ -0,0 +1,53 @@
+//-p infer-share
```

Comment:
Can you add a comment explaining what this test is doing and rename it to have a more informative name like "simple-dominate" or something?